internal/cmd/io_binary.go

```diff
@@ -61,7 +61,12 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (_ UpstreamIO, er
 	if err != nil {
 		return nil, err
 	}
-	defer waitPipe.Close()
+	defer func() {
+		err := waitPipe.Close()
+		if err != nil {
+			log.G(ctx).Errorf("error closing wait pipe: %s", waitPipePath)
```

Comment:
This doesn't log `err` itself, is it?

Reply:
Beat me to it, should chain `WithError` here

Reply:
oh lol, that's what I meant to do.